lib/blob.js

```diff
@@ -1044,17 +1193,957 @@ Blob.prototype.leaseContainer = function leaseContainer(name, options) {
+ *      lastModified: '...',       // The date/time the blob was last modified
+ *      metadata: '...'            // Name-value pairs that correspond to the user-defined metadata associated with this blob.
+ * }
+ * ```
+ */
+Blob.prototype.getBlobMetadata = function getBlobMetadata(container, blob, options) {
+  assert(typeof container === 'string', 'The name of the container must be specified and must be a string value.');
+  assert(typeof blob === 'string', 'The name of the blob must be specified and must be a string value.');
+
+  var query = {
+    comp: 'metadata'
+  }
+  var path = '/' + container + '/' + blob;
+  var headers = {};
+
+  utils.setConditionalHeaders(headers, options);
+
+  return this.request('HEAD', path, query, headers).then(function(response) {
+    if (response.statusCode !== 200) {
+      throw new Error("getBlobMetadata: Unexpected statusCode: " + response.statusCode);
+    }
+    var result = {
+      eTag: response.headers['etag'],
+      lastModified: response.headers['last-modified'],
+      metadata: utils.extractMetadataFromHeaders(response)
+    };
+    return result;
+  });
+};
+
+/**
+ * Sets metadata for the specified blob.
+ * Overwrites all existing metadata that is associated with that blob.
+ *
+ * @method setBlobMetadata
+ * @param {string} container - Name of the container
+ * @param {string} blob - Name of the blob
+ * @param {object} metadata - Mapping from metadata keys to values.
+ * @param {object} options - Options on the following form
+ * ```js
+ * {
+ *    ifModifiedSince: new Date(),      // Specify this to perform the operation only if the resource has been modified since the specified time. (optional)
+ *    ifUnmodifiedSince: new Date(),    // Specify this to perform the operation only if the resource has not been modified since the specified date/time. (optional)
+ *    ifMatch: '...',                   // ETag value. Specify this to perform the operation only if the resource's ETag matches the value specified. (optional)
+ *    ifNoneMatch: '...',               // ETag value. Specify this to perform the operation only if the resource's ETag does not match the value specified. (optional)
+ * }
+ *```
+ * @returns {Promise} A promise for an object on the form:
+ * ```js
+ * {
+ *      eTag: '...',               // The entity tag of the blob
+ *      lastModified: '...'        // The date/time the blob was last modified.
+ * }
+ * ```
+ */
+Blob.prototype.setBlobMetadata = function setBlobMetadata(container, blob, metadata, options) {
+  assert(typeof container === 'string', 'The name of the container must be specified and must be a string value.');
+  assert(typeof blob === 'string', 'The name of the blob must be specified and must be a string value.');
+
+  // Construct query string
+  var query = {
+    comp: 'metadata'
+  };
+  // Construct headers
+  var headers = {};
+
+  if (options) {
+    utils.setConditionalHeaders(headers, {
```

Comment:
Same where and everywhere else... Note I event support putting the `if (options) {..` part into `setConditionalHeaders` so we don't have to repeat it everywhere.